In [1]:
import numpy as np 
import pandas as pd 
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline
import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

/kaggle/input/202122-nba-season-active-nba-players/active_players_2.csv
/kaggle/input/nba-injuries-2010-2018/injuries_2010-2020.csv


In [2]:
df = pd.read_csv('/kaggle/input/nba-injuries-2010-2018/injuries_2010-2020.csv')

This dataset contains NBA players' injuries between 2010 and 2020.

First injury record was recorded on October 2010 and the last one on October 2020.

To find out the injury history of the active NBA players (21-22 season), we'll import the active players dataset and merge them.

In [3]:
df.tail()

Date    Team      Acquired  Relinquished  \
27100  2020-09-30  Lakers  Dion Waiters           NaN   
27101  2020-10-02    Heat           NaN   Bam Adebayo   
27102  2020-10-02    Heat           NaN  Goran Dragic   
27103  2020-10-02    Heat   Chris Silva           NaN   
27104  2020-10-06    Heat   Bam Adebayo           NaN   

                                                   Notes  
27100                                  activated from IL  
27101                                strained neck (DTD)  
27102  placed on IL with torn plantar fascia in left ...  
27103                                  activated from IL  
27104                                 returned to lineup

In [4]:
df.head()

Date     Team Acquired   Relinquished  \
0  2010-10-03    Bulls      NaN  Carlos Boozer   
1  2010-10-06  Pistons      NaN  Jonas Jerebko   
2  2010-10-06  Pistons      NaN  Terrico White   
3  2010-10-08  Blazers      NaN     Jeff Ayres   
4  2010-10-08     Nets      NaN    Troy Murphy   

                                               Notes  
0  fractured bone in right pinky finger (out inde...  
1      torn right Achilles tendon (out indefinitely)  
2  broken fifth metatarsal in right foot (out ind...  
3          torn ACL in right knee (out indefinitely)  
4             strained lower back (out indefinitely)

In [5]:
df.isnull().sum()

Date                0
Team                2
Acquired        17563
Relinquished     9545
Notes               0
dtype: int64

We will drop the Acquired and Relinquished columns and combine them under the column "Name".


In [6]:
df['Name'] = df[["Acquired","Relinquished"]].fillna('').sum(axis=1)

In [7]:
df = df.drop(columns=['Acquired', 'Relinquished'])

df.head()


Date     Team                                              Notes  \
0  2010-10-03    Bulls  fractured bone in right pinky finger (out inde...   
1  2010-10-06  Pistons      torn right Achilles tendon (out indefinitely)   
2  2010-10-06  Pistons  broken fifth metatarsal in right foot (out ind...   
3  2010-10-08  Blazers          torn ACL in right knee (out indefinitely)   
4  2010-10-08     Nets             strained lower back (out indefinitely)   

            Name  
0  Carlos Boozer  
1  Jonas Jerebko  
2  Terrico White  
3     Jeff Ayres  
4    Troy Murphy

In [8]:
df["Team"].value_counts()

Spurs           1163
Bucks           1068
Warriors        1060
Rockets         1058
Raptors         1044
Celtics         1040
Nets            1024
Heat            1023
Cavaliers       1001
Mavericks        992
Hawks            975
Nuggets          966
Lakers           959
Knicks           943
76ers            910
Wizards          875
Grizzlies        875
Timberwolves     860
Jazz             841
Magic            834
Pacers           831
Bulls            791
Suns             733
Kings            728
Hornets          719
Clippers         718
Thunder          717
Pistons          714
Blazers          695
Pelicans         576
Bobcats          369
Bullets            1
Name: Team, dtype: int64

Changing the team names to full names.

In [9]:
dict = {"Rockets":"Houston Rockets",
        "Magic":"Orlando Magic",
        "Nets":"Brooklyn Nets",
        "76ers":"Philadelphia Sixers",
        "Cavaliers":"Cleveland Cavaliers",
        "Kings":"Sacramento Kings",
        "Pacers":"Indiana Pacers",
        "Bucks":"Milwaukee Bucks",
        "Celtics":"Boston Celtics",
        "Pelicans":"New Orleans Pelicans",
        "Clippers":"Los Angeles Clippers",
        "Nuggets":"Denver Nuggets",
        "Wizards":"Washington Wizards",
        "Bullets":"Washington Bullets",
        "Thunder":"Oklahoma City Thunder",
        "Raptors":"Toronto Raptors",
        "Bulls":"Chicago Bulls",
        "Lakers":"Los Angeles Lakers",
        "Grizzlies":"Memphis Grizzlies",
        "Hawks":"Atlanta Hawks",
        "Heat":"Miami Heat",
        "Spurs":"San Antonio Spurs",
        "Mavericks":"Dallas Mavericks",
        "Jazz":"Utah Jazz",
        "Hornets":"Charlotte Hornets",
        "Bobcats":"Charlotte Bobcats",
        "Pistons":"Detroit Pistons",
        "Warriors":"Golden State Warriors",
        "Timberwolves":"Minnesota Timberwolves",
        "Suns":"Phoenix Suns",
        "Knicks":"New York Knicks",
        "Blazers":"Portland Trailblazers"
    
}

In [10]:
df["Team"].replace(dict, inplace=True)

df.tail(10)

Date                Team  \
27095  2020-09-12     Houston Rockets   
27096  2020-09-18      Boston Celtics   
27097  2020-09-22      Boston Celtics   
27098  2020-09-23          Miami Heat   
27099  2020-09-30          Miami Heat   
27100  2020-09-30  Los Angeles Lakers   
27101  2020-10-02          Miami Heat   
27102  2020-10-02          Miami Heat   
27103  2020-10-02          Miami Heat   
27104  2020-10-06          Miami Heat   

                                                   Notes            Name  
27095     placed on IL with neck spasms (out for season)  Tyson Chandler  
27096  placed on IL with strained right adductor (out...  Romeo Langford  
27097            surgery on right wrist (out for season)  Romeo Langford  
27098                              sore right knee (DTD)    Gabe Vincent  
27099                       strained left shoulder (DTD)     Bam Adebayo  
27100                                  activated from IL    Dion Waiters  
27101                                strained neck (DTD)     Bam Adebayo  
27102  placed on IL with torn plantar fascia in left ...    Goran Dragic  
27103                                  activated from IL     Chris Silva  
27104                                 returned to lineup     Bam Adebayo

In [11]:
df["Team"].value_counts()

San Antonio Spurs         1163
Milwaukee Bucks           1068
Golden State Warriors     1060
Houston Rockets           1058
Toronto Raptors           1044
Boston Celtics            1040
Brooklyn Nets             1024
Miami Heat                1023
Cleveland Cavaliers       1001
Dallas Mavericks           992
Atlanta Hawks              975
Denver Nuggets             966
Los Angeles Lakers         959
New York Knicks            943
Philadelphia Sixers        910
Memphis Grizzlies          875
Washington Wizards         875
Minnesota Timberwolves     860
Utah Jazz                  841
Orlando Magic              834
Indiana Pacers             831
Chicago Bulls              791
Phoenix Suns               733
Sacramento Kings           728
Charlotte Hornets          719
Los Angeles Clippers       718
Oklahoma City Thunder      717
Detroit Pistons            714
Portland Trailblazers      695
New Orleans Pelicans       576
Charlotte Bobcats          369
Washington Bullets           1
Name: Te

Changing the order of the rows

In [12]:
df = df[["Name", "Team", "Date", "Notes"]]

df.tail()

Name                Team        Date  \
27100  Dion Waiters  Los Angeles Lakers  2020-09-30   
27101   Bam Adebayo          Miami Heat  2020-10-02   
27102  Goran Dragic          Miami Heat  2020-10-02   
27103   Chris Silva          Miami Heat  2020-10-02   
27104   Bam Adebayo          Miami Heat  2020-10-06   

                                                   Notes  
27100                                  activated from IL  
27101                                strained neck (DTD)  
27102  placed on IL with torn plantar fascia in left ...  
27103                                  activated from IL  
27104                                 returned to lineup

Changing the "Date" column to datetime to be able to format the dates.

In [13]:
df['Date'] = pd.to_datetime(df.Date)

df['Date'] = df['Date'].dt.strftime('%d/%m/%Y')

df.tail()

Name                Team        Date  \
27100  Dion Waiters  Los Angeles Lakers  30/09/2020   
27101   Bam Adebayo          Miami Heat  02/10/2020   
27102  Goran Dragic          Miami Heat  02/10/2020   
27103   Chris Silva          Miami Heat  02/10/2020   
27104   Bam Adebayo          Miami Heat  06/10/2020   

                                                   Notes  
27100                                  activated from IL  
27101                                strained neck (DTD)  
27102  placed on IL with torn plantar fascia in left ...  
27103                                  activated from IL  
27104                                 returned to lineup

Dropping the team column as the active player dataset already contains it.

In [14]:
df = df.drop(columns="Team")
df.tail()

Name        Date  \
27100  Dion Waiters  30/09/2020   
27101   Bam Adebayo  02/10/2020   
27102  Goran Dragic  02/10/2020   
27103   Chris Silva  02/10/2020   
27104   Bam Adebayo  06/10/2020   

                                                   Notes  
27100                                  activated from IL  
27101                                strained neck (DTD)  
27102  placed on IL with torn plantar fascia in left ...  
27103                                  activated from IL  
27104                                 returned to lineup

Importing active player dataset for 21-22 season.

This dataset includes Name, Team, Position, Age, Height, Weight, College and Salary information of the 2021-22 season NBA players.

We will only use Name, Team and Position information for now.

In [15]:
df_act = pd.read_csv("/kaggle/input/202122-nba-season-active-nba-players/active_players_2.csv")

df_act = df_act[["Name","Team","Position"]]

df_act.head()



Name            Team Position
0  Juhann Begarin  Boston Celtics       SG
1    Jaylen Brown  Boston Celtics       SG
2       Kris Dunn  Boston Celtics       PG
3  Carsen Edwards  Boston Celtics       PG
4      Tacko Fall  Boston Celtics        C

Merging the datasets by performing a left merge on Name.

In [16]:
result_df = pd.merge(df_act, df, how= "left", on=["Name"])
result_df = result_df[result_df['Date'].notna()]

result_df.tail()

Name       Team Position        Date  \
9489  Hassan Whiteside  Utah Jazz        C  13/01/2020   
9490  Hassan Whiteside  Utah Jazz        C  07/02/2020   
9491  Hassan Whiteside  Utah Jazz        C  11/02/2020   
9492  Hassan Whiteside  Utah Jazz        C  08/08/2020   
9493  Hassan Whiteside  Utah Jazz        C  11/08/2020   

                                    Notes  
9489                   returned to lineup  
9490               bruised left leg (DTD)  
9491                   returned to lineup  
9492  placed on IL with strained left hip  
9493                    activated from IL

Clearing Null entries

In [17]:
result_df.head(30)

Name            Team Position        Date  \
1   Jaylen Brown  Boston Celtics       SG  11/01/2017   
2   Jaylen Brown  Boston Celtics       SG  13/01/2017   
3   Jaylen Brown  Boston Celtics       SG  13/02/2017   
4   Jaylen Brown  Boston Celtics       SG  24/02/2017   
5   Jaylen Brown  Boston Celtics       SG  06/12/2017   
6   Jaylen Brown  Boston Celtics       SG  08/12/2017   
7   Jaylen Brown  Boston Celtics       SG  21/12/2017   
8   Jaylen Brown  Boston Celtics       SG  23/12/2017   
9   Jaylen Brown  Boston Celtics       SG  27/12/2017   
10  Jaylen Brown  Boston Celtics       SG  27/12/2017   
11  Jaylen Brown  Boston Celtics       SG  31/12/2017   
12  Jaylen Brown  Boston Celtics       SG  09/03/2018   
13  Jaylen Brown  Boston Celtics       SG  11/03/2018   
14  Jaylen Brown  Boston Celtics       SG  25/03/2018   
15  Jaylen Brown  Boston Celtics       SG  30/04/2018   
16  Jaylen Brown  Boston Celtics       SG  03/05/2018   
17  Jaylen Brown  Boston Celtics       SG  01/11/2018   
18  Jaylen Brown  Boston Celtics       SG  03/11/2018   
19  Jaylen Brown  Boston Celtics       SG  26/11/2018   
20  Jaylen Brown  Boston Celtics       SG  06/12/2018   
21  Jaylen Brown  Boston Celtics       SG  12/12/2018   
22  Jaylen Brown  Boston Celtics       SG  14/12/2018   
23  Jaylen Brown  Boston Celtics       SG  01/04/2019   
24  Jaylen Brown  Boston Celtics       SG  07/04/2019   
25  Jaylen Brown  Boston Celtics       SG  30/10/2019   
26  Jaylen Brown  Boston Celtics       SG  07/11/2019   
27  Jaylen Brown  Boston Celtics       SG  30/12/2019   
28  Jaylen Brown  Boston Celtics       SG  31/12/2019   
29  Jaylen Brown  Boston Celtics       SG  03/01/2020   
30  Jaylen Brown  Boston Celtics       SG  16/01/2020   

                                          Notes  
1                    sprained right ankle (DTD)  
2                            returned to lineup  
3   placed on IL with strained right hip flexor  
4                             activated from IL  
5      placed on IL with right eye inflammation  
6                             activated from IL  
7                      sore left Achilles (DTD)  
8                            returned to lineup  
9                                  placed on IL  
10                        sore right knee (DTD)  
11                            activated from IL  
12                concussion (out indefinitely)  
13                 placed on IL with concussion  
14                            activated from IL  
15               strained right hamstring (DTD)  
16                           returned to lineup  
17            placed on IL with sore right foot  
18                            activated from IL  
19         placed on IL with bruised lower back  
20                            activated from IL  
21                                illness (DTD)  
22                           returned to lineup  
23          placed on IL with lower back spasms  
24                            activated from IL  
25                    placed on IL with illness  
26                            activated from IL  
27                                illness (DTD)  
28                    placed on IL with illness  
29                            activated from IL  
30       placed on IL with sprained right thumb

9493 injury records of active players are available

Looking for a specific player's injury records for the last 10 years

In [18]:
result_df[result_df["Name"] == "Paul George"]

Name                  Team Position        Date  \
5480  Paul George  Los Angeles Clippers       SG  28/11/2010   
5481  Paul George  Los Angeles Clippers       SG  13/12/2010   
5482  Paul George  Los Angeles Clippers       SG  17/12/2010   
5483  Paul George  Los Angeles Clippers       SG  29/12/2010   
5484  Paul George  Los Angeles Clippers       SG  12/01/2013   
5485  Paul George  Los Angeles Clippers       SG  13/01/2013   
5486  Paul George  Los Angeles Clippers       SG  17/04/2013   
5487  Paul George  Los Angeles Clippers       SG  09/04/2014   
5488  Paul George  Los Angeles Clippers       SG  01/08/2014   
5489  Paul George  Los Angeles Clippers       SG  29/10/2014   
5490  Paul George  Los Angeles Clippers       SG  05/04/2015   
5491  Paul George  Los Angeles Clippers       SG  24/03/2016   
5492  Paul George  Los Angeles Clippers       SG  13/04/2016   
5493  Paul George  Los Angeles Clippers       SG  12/11/2016   
5494  Paul George  Los Angeles Clippers       SG  14/11/2016   
5495  Paul George  Los Angeles Clippers       SG  18/11/2016   
5496  Paul George  Los Angeles Clippers       SG  23/11/2016   
5497  Paul George  Los Angeles Clippers       SG  25/11/2016   
5498  Paul George  Los Angeles Clippers       SG  04/12/2016   
5499  Paul George  Los Angeles Clippers       SG  07/12/2017   
5500  Paul George  Los Angeles Clippers       SG  11/12/2017   
5501  Paul George  Los Angeles Clippers       SG  29/12/2017   
5502  Paul George  Los Angeles Clippers       SG  29/12/2017   
5503  Paul George  Los Angeles Clippers       SG  31/12/2017   
5504  Paul George  Los Angeles Clippers       SG  09/05/2018   
5505  Paul George  Los Angeles Clippers       SG  28/12/2018   
5506  Paul George  Los Angeles Clippers       SG  30/12/2018   
5507  Paul George  Los Angeles Clippers       SG  28/02/2019   
5508  Paul George  Los Angeles Clippers       SG  05/03/2019   
5509  Paul George  Los Angeles Clippers       SG  10/04/2019   
5510  Paul George  Los Angeles Clippers       SG  14/04/2019   
5511  Paul George  Los Angeles Clippers       SG  22/10/2019   
5512  Paul George  Los Angeles Clippers       SG  14/11/2019   
5513  Paul George  Los Angeles Clippers       SG  04/01/2020   
5514  Paul George  Los Angeles Clippers       SG  05/01/2020   
5515  Paul George  Los Angeles Clippers       SG  09/01/2020   
5516  Paul George  Los Angeles Clippers       SG  10/01/2020   
5517  Paul George  Los Angeles Clippers       SG  30/01/2020   
5518  Paul George  Los Angeles Clippers       SG  21/02/2020   
5519  Paul George  Los Angeles Clippers       SG  22/02/2020   
5520  Paul George  Los Angeles Clippers       SG  24/02/2020   

                                                  Notes  
5480                                       placed on IL  
5481                                  activated from IL  
5482                                       placed on IL  
5483                                  activated from IL  
5484                          placed on IL with illness  
5485                                  activated from IL  
5486                                                DNP  
5487                                         rest (DNP)  
5488  surgery on right leg to repair fractured tibia...  
5489  placed on IL recovering from surgery on right ...  
5490                                  activated from IL  
5491                      bruised lower right leg (DTD)  
5492                                         rest (DTD)  
5493              placed on IL with sprained left ankle  
5494                                  activated from IL  
5495                  placed on IL with sore left ankle  
5496                                  activated from IL  
5497                       sore back / left ankle (DTD)  
5498                                 returned to lineup  
5499               placed on IL with bruised right calf  
5500                                  activated from IL  
5501                   placed on IL with sore le

In [19]:
result_df.describe()

Name                Team Position        Date              Notes
count         9280                9280     9280        9280               9280
unique         344                  30        7        1734               1552
top     Kevin Love  Los Angeles Lakers        C  12/04/2017  activated from IL
freq           139                 715     1943          35               2480

In [20]:
result_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9280 entries, 1 to 9493
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   Name      9280 non-null   object
 1   Team      9280 non-null   object
 2   Position  9280 non-null   object
 3   Date      9280 non-null   object
 4   Notes     9280 non-null   object
dtypes: object(5)
memory usage: 435.0+ KB


In [21]:
filename = 'Injury_History.csv'

result_df.to_csv(filename,index=False)

print('Saved file: ' + filename)

Saved file: Injury_History.csv
